pkg/server/server.go

```diff
@@ -238,6 +290,16 @@ func (h *handler) authenticateEndpoint(w http.ResponseWriter, req *http.Request)
 	} else if userMapping, exists := h.lowercaseUserMap[arnLower]; exists {
 		username = userMapping.Username
 		groups = userMapping.Groups
+	} else if nodeMapping, exists := h.lowercaseNodeMap[arnLower]; exists {
```

Comment:
If the config contains Node's IAM role as a RoleMapping then this block won't be executed right? Do we want that to happen?

Reply:
Right.  I'm not sure if we care which takes precedence, but if you have an argument one way or the other I'm all ears.

Reply:
If the authenticated user is an EC2 instance, then NodeMapping should override.

You can determine whether the authenticated user is an EC2 instance or not by inspecting the UserId field in GetCallerIdentity response. It should be in the format "role-id:ec2-instance-id" as per https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_variables.html#principaltable